pkg/metrics/task_metrics_mock.go

```diff
@@ -0,0 +1,158 @@
+// Code generated by mockery v2.20.0. DO NOT EDIT.
```

Comment:
I like it! Could you describe in the readme how to generate mocks with Mockery? Perhaps we could standardize our mocks using mockery